src/pharmpy/modeling/data.py

```diff
@@ -227,8 +231,12 @@ def get_mdv(model):
         try:
             label = model.dataset.pharmpy.labels_by_type[ColumnType.DOSE]
         except KeyError:
-            raise DatasetError('Could not find EVID or AMT column in dataset')
+            label = model.dataset.pharmpy.labels_by_type[ColumnType.DV]
+
+    data = model.dataset[label].astype('float64').squeeze()
 
-    dose = model.dataset[label].astype('float64').squeeze()
-    mdv = dose.where(dose == 0, other=1).astype('int64').rename('MDV')
+    if label[0] == 'DV':
```

Comment:
You shouldn't assume that the column name is 'DV' here. Handle this case directly in the except and then return.

Reply:
Now it looks better. Thanks!